_posts/2019-05-21-i18n-messages.adoc

```diff
@@ -0,0 +1,327 @@
+---
+compatible_cuba_versions: 7.0+
+compatible_java_versions: 8+
+project_id: cuba-petclinic-i18n-messages
+permalink: localization-in-cuba-applications
+---
+= Localization in CUBA applications
+:showtitle:
+:sectlinks:
+:sectanchors:
+:page-navtitle: Localization in CUBA applications
+:page-excerpt: In this guide the localization features of a CUBA application are explained
+:page-root: ../../../
+:project_id: cuba-petclinic-i18n-messages
+:java_version: 1.8
+:cuba_version: 7.0
+
+
+CUBA applications have build-in support for localized messages for different languages. Translations are externalized in
+properties files to have a clear separation between the application code and the translated messages.
+
+
+== What Will be Built
+
+This guide enhances the https://github.com/cuba-platform/cuba-petclinic[CUBA petclinic] example to show how localized messages within the CUBA application can be defined and used:
+
+* the application is also available in German and Russian language
+* dynamic data as automatically generated descriptions for visits are localized
+* provide german translation for User browse platform screen
+
+=== Final Application
+
+++++
+<a href="https://raw.githubusercontent.com/cuba-guides/cuba-petclinic-i18n-messages/master/img/adjustments-cuba-petclinic-overview.gif"><img src="https://raw.githubusercontent.com/cuba-guides/cuba-petclinic-i18n-messages/master/img/login-screen.png"/></a>
+++++
+
+include::includes/guide_requirements.adoc[]
+
+
+include::includes/petclinic_introduction.adoc[]
+
+
+== Where Translations are Stored in a CUBA Application
+
+CUBA follows a common pattern for translations in Java based applications. Messages for different languages / locales are stores next to the source-code within the same package in `.properties` files.
+
+For every language and every package there exists a file containing information on how to translate a given text into the target language. The language files contain a suffix according to the locale code of the language (e.g. `_ru` for russian, `_de` for german).
```

Comment:
I think you should mention "default" files without suffixes, because in fact you show them below. A default file is used if there is no file with suffix matching user's locale. 
When using localization dialogs in Studio, it puts messages for the first selected locale to the default file and others to the files with suffixes. Maybe it's worth mentioning. Also I think would be great to give some recommendations on choosing what locale should be "default", i.e. corresponding to default files, like in the docs: "If the application is international, it is reasonable to use default files for the language of the application primary audience or for the English language, so that the messages from these default files are displayed to the user if the messages in the required language are not found." Pls feel free to suggest your way.

Reply:
👍